test/test_ensemblrest.py

```diff
@@ -68,691 +67,734 @@
-        
-        #add function name to the class methods
+        self.EnsEMBL.__dict__["notImplemented"] = self.EnsEMBL.register_api_func(
+            "notImplemented",
+            ensemblrest.ensembl_config.ensembl_api_table
+        )
+
+        # Set __doc__ for generic class method
+        self.EnsEMBL.__dict__["notImplemented"].__doc__ = ensemblrest.ensemblrest\
+            .ensembl_api_table["notImplemented"]["doc"]
+
+        # add function name to the class methods
         self.EnsEMBL.__dict__["notImplemented"].__name__ = "notImplemented"
-        
+
         # call the new function and deal with the exception
-        self.assertRaises(NotImplementedError, self.EnsEMBL.notImplemented, id='ENSG00000157764') 
-        
+        self.assertRaises(NotImplementedError, self.EnsEMBL.notImplemented, id='ENSG00000157764')
+
     def __something_bad(self, curl_cmd, last_response):
         """A function to test 'something bad' message"""
-        
+
         # execute the curl cmd an get data as a dictionary
         reference = jsonFromCurl(curl_cmd)
-        
+
         # create a fake request.Response class
         class FakeResponse():
             def __init__(self, response):
                 self.headers = response.headers
                 self.status_code = 400
                 self.text = """{"error":"something bad has happened"}"""
-                
-        #instantiate a fake response
+
+        # instantiate a fake response
         fakeResponse = FakeResponse(last_response)
         test = self.EnsEMBL.parseResponse(fakeResponse)
-        
+
         # testing values
         self.assertDictEqual(reference, test)
         self.assertGreaterEqual(self.EnsEMBL.last_attempt, 1)
-    
+
     def test_SomethingBad(self):
         """Deal with the {"error":"something bad has happened"} message"""
-        
+
         # get the curl cmd from ensembl site:
         curl_cmd = "curl 'http://rest.ensembl.org/archive/id/ENSG00000157764?' -H 'Content-type:application/json'"
-        
+
         # get a request
         self.EnsEMBL.getArchiveById(id="ENSG00000157764")
-        
+
         # retrieve last_reponse
         last_response = self.EnsEMBL.last_response
-        
+
         # call generic function
         self.__something_bad(curl_cmd, last_response)
-        
+
     def test_SomethingBadPOST(self):
         """Deal with the {"error":"something bad has happened"} message using a POST method"""
-        
-        curl_cmd = """curl 'http://rest.ensembl.org/lookup/id' -H 'Content-type:application/json' -H 'Accept:application/json' -X POST -d '{ "ids" : ["ENSG00000157764", "ENSG00000248378" ] }'"""
-      
+
+        curl_cmd = """curl 'http://rest.ensembl.org/lookup/id' -H 'Content-type:application/json' """\
+                   """-H 'Accept:application/json' -X POST -d '{ "ids" : ["ENSG00000157764", "ENSG00000248378" ] }'"""
```

Comment:
This can be expressed in a more `pep8` way as:
```python
curl_cmd = (
        """curl 'http://rest.ensembl.org/lookup/id' -H 'Content-type:"""
        """application/json' -H 'Accept:application/json' -X POST -d '"""
        """{ "ids" : ["ENSG00000157764", "ENSG00000248378" ] }'"""
        )
```
You will have less than 80 column and I think it will be more readable (I learn this after writing those test)

Reply:
I'm with you on this one. I just quickly went through all of them. Feel free to change any on these to be more readable. It was just way too many of them to fix each one properly this time.

Reply:
@bunop I believe, as a contributor you should have write access to my repo as well.